### PR TITLE
[feature|cleaning] adding forrgotten password methodology

### PR DESCRIPTION
**Pull request by profxed**
cleaning up accountConfirmation related files and adding forgotten password methodology.

Please have a look and let mo know if there is anything I can change.

### CODE REVIEW

File: src/routes/Routes.tsx [hunk 1]
```diff
@@ -42,6 +42,13 @@ const Routes: React.FC = () => (
       <Route exact={true} path="/about" component={AboutPage} />
       <Route exact={true} path="/login" component={SignIn} />
       <Route exact={true} path="/register" component={Register} />
+      {/* // ? what to do here? // */}
+      {/* <Route exact={true} path="/password" component={Password} /> */}
```
Comment (remarcmij) [inline src/routes/Routes.tsx:46]: I'm not quite sure why we need to have a single Password component to deal with both PasswordChange and PassewordReset,  but could have two different routes (one protected and the other one unprotected) using the same componet. So this one could be `/reset` and the protected one `/password`. Or am I missing something.

Comment (profxed) [reply src/routes/Routes.tsx:46]: I was thinking of the user who can't change his password from the inside but who can change it using `I forgot my password`, and that lead me to the point where a single Password component can do both jobs _(the reset)_ and _(the change)_, also same for the backend and from that I aimed to have as less code as possible,And as DRY code as possible.

do you think I have to make an addition route as `/reset`? And what do you think about my Idea, I mean for the future is that a good thing?

**Comment by remarcmij** [comment]
I will try and draw out the conversation between client and serve in a diagram for the various scenarios tomorrow. For me it is difficult to understand what happens in detail without some visual aid.

With respect to making your code DRY, there is sometimes a trade-off between DRYnesh and simplicity and maintainability of code. For instance, there is also a principle that a function (or in our case, a component) should do one thing only. If this turns out to be the more important consideration then you may need to compromise on DRYness. I will come back on this when I have a better understanding of your code tomorrow.

**Comment by profxed** [comment]
perfect looking forward.

File: src/routes/Routes.tsx (hunk 1 above)
Comment (remarcmij) [reply src/routes/Routes.tsx:46]: After going through your code I had no difficulty with using the same component(s) for password change and reset in the frontend, but I found it better to separate out these processes in the backend. See my PRs on this branch.

**Pull request merged**